Duplicati/Library/Backend/pCloud/pCloudBackend.cs

```diff
@@ -459,21 +493,48 @@ public Task CreateFolderAsync(CancellationToken cancellationToken)
 
         if (pCloudErrorList.ErrorMessages.TryGetValue(createFolderResponse.result, out var message))
             throw new Exception(message);
-        /*
-         * If no error code is matched, result was == 0 so it successfully created the folder
-         * We must cache the folderID, as that will be used to upload files, reason being that
-         * for oauth clients, pCloud creates a scoped folder Applications/ApplicationName
-         * and the folders created with /createfolderifnotexists api are created under that folder.
-         *
-         * However, when uploading files, the folderid is required (and surprisingly if not provided
-         * it will create the file outside the scoped folder, at the root of the drive)
-         */
-        if (createFolderResponse.result == 0) _CachedFolderID = createFolderResponse.metadata?.folderid;
+   
+        return createFolderResponse is { result: 0, metadata.folderid: var id }
+            ? Task.FromResult(id)
+            : throw new Exception(Strings.pCloudBackend.FailedWithUnexpectedErrorCode("createfolder", createFolderResponse.result));
+    }
 
-        return Task.CompletedTask;
+    /// <summary>
+    /// Returns the fileID by listing the folder and searching for the filename & metadata.
+    ///
+    /// For operations such as delete/getfilelink using the fileID is more reliable than using direct path/filename concatenation.
+    /// </summary>
+    /// <param name="name">The filename</param>
+    /// <returns></returns>
+    /// <exception cref="FileNotFoundException"></exception>
+    private Task<ulong> GetFileId(string name)
```

Comment:
This should take a cancellationToken

Reply:
Latest commit has implemented this alongside with all cancellation token propagation.